examples/pub-sub-channel-state/react/src/app/layout.tsx

```diff
@@ -0,0 +1,37 @@
+"use client";
+
+import { Inter } from "next/font/google";
+import { Realtime } from 'ably';
+import { AblyProvider, ChannelProvider } from 'ably/react';
+import { nanoid } from 'nanoid';
+import '../../styles/styles.css';
+import { useEffect, useState } from "react";
+
+const inter = Inter({ subsets: ["latin"] });
+const client = new Realtime({key: process.env.NEXT_PUBLIC_ABLY_KEY, clientId: nanoid()});
+
+export default function RootLayout({
+  children,
+}: Readonly<{
+  children: React.ReactNode;
+}>) {
+  const [channelName, setChannelName] = useState('pub-sub-channel-state');
+
+  useEffect(() => {
+    const urlParams = new URLSearchParams(window.location.search);
```

Comment:
Not sure you need to do this in `useEffect`, I think you can simply put this code outside the component

Reply:
<img width="1018" alt="Screenshot 2025-02-24 at 10 04 29" src="https://github.com/user-attachments/assets/72e3b66a-ceed-44aa-803c-874a510afd68" />

If I do it this way without a `useEffect`, I get a rendering issue. I think it's an issue with Next and SSR.